control/tests/statesp_test.py

```diff
@@ -751,6 +763,58 @@ def test_horner(self, sys322):
             np.squeeze(sys322.horner(1.j)),
             mag[:, :, 0] * np.exp(1.j * phase[:, :, 0]))
 
+    @pytest.mark.parametrize('x',
+        [[1, 1], [[1], [1]], np.atleast_2d([1,1]).T])
+    @pytest.mark.parametrize('u', [0, 1, np.atleast_1d(2)])
+    def test_dynamics_and_output_siso(self, x, u, sys121):
+        assert_array_almost_equal(
+            sys121.dynamics(0, x, u),
+            sys121.A.dot(x).reshape((-1,)) + sys121.B.dot(u).reshape((-1,)))
+        assert_array_almost_equal(
+            sys121.output(0, x, u),
+            sys121.C.dot(x).reshape((-1,)) + sys121.D.dot(u).reshape((-1,)))
+
```

Comment:
Add a test for the case where you only pass two arguments?  Could just be added to test_dynamics_and_output_siso as a new case (where you ignore u => a bit wasteful, but easy to add).

Reply:
wanted to do something like that, just couldn't think of how to do it after a few hours of coding : ) Thanks for the suggestion.